receiver/signalfxreceiver/signalfxv2_to_metricdata.go

```diff
@@ -0,0 +1,238 @@
+	sfxpb "github.com/signalfx/com_signalfx_metrics_protobuf"
+	"go.uber.org/zap"
+)
+
+var (
+	errSFxNilDatum                   = errors.New("nil datum value for data-point")
+	errSFxMetricTypeEnumNotSupported = errors.New("data-point of MetricType_ENUM type is not supported")
+
+	errSFxUnexpectedInt64DatumType   = errors.New("datum value type of int64 is unexpected")
+	errSFxUnexpectedFloat64DatumType = errors.New("datum value type of float64 is unexpected")
+	errSFxUnexpectedStringDatumType  = errors.New("datum value type of string is unexpected")
+	errSFxStringDatumNotNumber       = errors.New("datum string cannot be parsed to a number")
+	errSFxNoDatumValue               = errors.New("no datum value present for data-point")
+)
+
+// SignalFxV2ToMetricsData converts SignalFx proto data points to
+// consumerdata.MetricsData. Returning the converted data and the number of
+// dropped time series.
+func SignalFxV2ToMetricsData(
+	logger *zap.Logger,
+	sfxDataPoints []*sfxpb.DataPoint,
+) (*consumerdata.MetricsData, int) {
+
+	// TODO: not optimized at all, basically regenerating everything for each
+	// 	data point.
+	numDroppedTimeSeries := 0
+	md := &consumerdata.MetricsData{}
+	metrics := make([]*metricspb.Metric, 0, len(sfxDataPoints))
+	for _, sfxDataPoint := range sfxDataPoints {
+		if sfxDataPoint == nil {
+			// TODO: Log or metric for this odd ball?
+			continue
+		}
+
+		// First check if the type is convertible and the data point is consistent.
+		metricType, err := convertType(sfxDataPoint)
+		if err != nil {
+			numDroppedTimeSeries++
+			logger.Debug("SignalFx data-point type conversion error",
+				zap.Error(err),
+				zap.String("metric", sfxDataPoint.GetMetric()))
+			continue
+		}
+		point, err := buildPoint(sfxDataPoint, metricType)
+		if err != nil {
+			numDroppedTimeSeries++
+			logger.Debug("SignalFx data-point datum conversion error",
+				zap.Error(err),
+				zap.String("metric", sfxDataPoint.GetMetric()))
+			continue
+		}
+
+		labelKeys, labelValues := buildLabelKeysAndValues(sfxDataPoint.Dimensions)
+		descriptor := buildDescriptor(sfxDataPoint, labelKeys, metricType)
+		ts := &metricspb.TimeSeries{
+			// TODO: StartTimestamp can be set if each cumulative time series are
+			//  	tracked but right now it is not clear if it brings benefits.
+			//		Perhaps as an option so cost is "pay for play".
+			LabelValues: labelValues,
+			Points:      []*metricspb.Point{point},
+		}
+		metric := &metricspb.Metric{
+			MetricDescriptor: descriptor,
+			Timeseries:       []*metricspb.TimeSeries{ts},
+		}
+		metrics = append(metrics, metric)
+	}
+
+	md.Metrics = metrics
+	return md, numDroppedTimeSeries
+}
+
+func convertType(
+	sfxDataPoint *sfxpb.DataPoint,
+) (descType metricspb.MetricDescriptor_Type, err error) {
+
+	// Combine metric type with the actual data point type
+	sfxMetricType := sfxDataPoint.GetMetricType()
+	sfxDatum := sfxDataPoint.Value
+	if sfxDatum == nil {
+		return descType, errSFxNilDatum
```

Comment:
What's the value of `descType` at this point? Is it the zero value? Perhaps better to specify it explicitly to make it clear?

Reply:
Yes, the zero value - will add explicitly.